### PR TITLE
Add parameter for specifying what to do with elements that already exist within the mount.

### PR DESCRIPTION
**Pull request by AlterionX**
The last part of #235.

- Allow Seed to take over the children of the mount. This can be useful for SSR. This is partially related to #232, which is also related to #223.
- Warns of possibly questionable behavior when attempting to accomplish the above maneuver as it is now (see the doc comment for `takeover_mount` in `AppBuilder`). Due to this behavior and backwards compatibility, it is currently an opt in functionality.

### CODE REVIEW

**Comment by AlterionX** [comment]
I'll close #235 after this gets merged.

File: src/vdom.rs [hunk 1]
```diff
@@ -195,29 +197,67 @@ impl<Ms, Mdl, ElC: View<Ms> + 'static, GMs: 'static> App<Ms, Mdl, ElC, GMs> {
...
+        // Map the DOM's elements onto the virtual DOM if requested to takeover.
+        if mount_type == MountType::Takeover {
+            // Construct a vdom from the root element. Subsequently strip the workspace so that we
+            // can recreate it later - this is a kind of simple way to avoid missing nodes (but
+            // not entirely correct).
+            // TODO: 1) Optimize by utilizing a patching strategy instead of recreating the workspace
+            // TODO: nodes.
+            // TODO: 2) Watch out for elements that should not be recreated, such as script nodes,
+            // TODO: and other, similar things. For now, leave the warning in the builder's
+            // TODO: documentation.
```
Comment (MartinKavik) [inline src/vdom.rs]: Could you move this comment block into a new GitHub issue and just add a shorter comment with todo and link to that issue here?
So we don't forget to fix it and you can describe the problem / solution in more details (if you want) while you have details still in your head :)

File: src/vdom.rs [hunk 2]
```diff
@@ -195,29 +197,67 @@ impl<Ms, Mdl, ElC: View<Ms> + 'static, GMs: 'static> App<Ms, Mdl, ElC, GMs> {
...
+
         self.setup_window_listeners();
         patch::setup_input_listeners(&mut new);
         patch::attach_listeners(&mut new, &self.mailbox());
 
-        websys_bridge::assign_ws_nodes_to_el(&util::document(), &mut new);
+        // Recreate the needed nodes. Only do this if requested to takeover the mount point since
+        // it should only be needed here.
+        if mount_type == MountType::Takeover {
+            // TODO: Refer the Optimization TODO.
```
Comment (MartinKavik) [inline src/vdom.rs]: So.. what's the TODO?  Can I remove this comment once the optimization issue is resolved?

File: src/vdom/builder.rs [hunk 3]
```diff
@@ -38,31 +38,64 @@ impl MountPoint for web_sys::HtmlElement {
...
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
 pub enum UrlHandling {
     PassToRoutes,
     None,
     // todo: Expand later, as-required
 }
 
+/// Describes the handling of elements already present in the mount element.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub enum MountType {
```
Comment (MartinKavik) [inline src/vdom/builder.rs]: Add `MountType` into prelude please:
Replace:
https://github.com/David-OConnor/seed/blob/179e11d3238ab68eb265dc38646099da01ad3631/src/lib.rs#L100
With:
```rust
vdom::{Init, UrlHandling, MountType},
```

**Comment by MartinKavik** [comment]
I tested it with a project based on webpack quickstart with and without prerendering and everything seems to work. Good job. I think we can merge it once the review comments are resolved.

File: src/vdom.rs (hunk 1 above)
Comment (AlterionX) [reply src/vdom.rs]: Got it.

File: src/vdom.rs (hunk 2 above)
Comment (AlterionX) [reply src/vdom.rs]: Yes. It's the TODO referenced in the other comment.

File: src/vdom/builder.rs (hunk 3 above)
Comment (AlterionX) [reply src/vdom/builder.rs]: Got it.

**Comment by AlterionX** [comment]
I'll fix the mentioned issues soon.

File: src/vdom.rs (hunk 1 above)
Comment (AlterionX) [reply src/vdom.rs]: #277

File: src/lib.rs [hunk 4]
```diff
@@ -8,7 +8,7 @@ pub use crate::{
     fetch::{Method, Request},
     routing::{push_route, Url},
     util::{body, cookies, document, error, history, html_document, log, update, window},
-    vdom::{App, AppBuilder},
+    vdom::{App, AppBuilder, MountType},
```
Comment (MartinKavik) [inline src/lib.rs]: Why did you add it here instead of `prelude`?

Comment (AlterionX) [reply src/lib.rs]: Oh, oops. My bad. Must have been tired yesterday.

**Comment by David-OConnor** [comment]
Adding this to the guide.
